framework/src/modules/chain/modules/rounds.js

```diff
@@ -62,6 +64,128 @@ class Rounds {
 	}
 }
 
+__private.updateSendersRoundInformationWithAmountForTransactions = function(
+	round,
+	transactions,
+	forwardTick,
+	tx
+) {
+	return transactions.map(transaction => {
+		const value = new Bignumber(transaction.amount).plus(transaction.fee);
+		const valueToUpdate = forwardTick ? `-${value}` : `${value}`;
+		return self.createRoundInformationWithAmount(
+			transaction.senderId,
+			round,
+			valueToUpdate,
+			tx
+		);
+	});
+};
+
+__private.updateRecipientsRoundInformationWithAmountForTransactions = function(
+	round,
+	transactions,
+	forwardTick,
+	tx
+) {
+	// For IN_TRANSFER TRANSACTION, we first need to get the authorId to get the recipient
+	return Promise.all(
+		transactions
+			.filter(transaction => transaction.type === transactionTypes.IN_TRANSFER)
+			.map(transaction =>
+				// Check the owner of the dapp for recipient account
+				library.storage.entities.Transaction.getOne(
+					{
+						id: transaction.asset.inTransfer.dappId,
```

Comment:
You can first map to get the ids of the transactions and then use `id_in` to avoid running multiple queries against active transaction connection.